add server side sampling docs (#5315)
getsentry/sentry-docs: src/docs/product/data-management-settings/server-side-sampling/current-limitations.mdx

---
title: Current Limitations
sidebar_order: 20
description: "Learn about the current limitions of server-side sampling."
---

> 

Author: ```suggestion

<Note>

This feature is in beta and is only available if your organization is participating in its limited release. Features in beta are still in-progress and may have bugs. We recognize the irony. If you’re interested in joining this limited release, please reach out at [abc@sentry.io](mailto:abc@sentry.io).

</Note>

```

Following are the current known limitations and functionality specifics that can create challenges when you're writing server-side sampling rules.

## Supported SDKs & Versions

Server-side sampling is currently only available for the following SDK's:

> - Python: 1.4.0 or later
> - JavaScript: 6.13.0 or later

Author: ```suggestion
- Python: 1.84.0 or later
- JavaScript: 7.5.1 or later
```

- Apple Cocoa: 7.3.0 or later
- Java and Android: 5.2.0 or later

<!-- TO DO version numbers need to be confirmed/updated -->

If your application relies on any Sentry SDK that isn't specified above, then you won’t be able to use server-side sampling.

## Trace rules can only select based on the attributes of the initial transaction

When sampling traces the decision on whether to keep or throw away a trace needs to be taken when the first transaction of the trace is seen. This in turn means that for transaction traces only the attributes of the first transaction can participate in the sampling decision.

> In conclusion all trace rules use the attributes (release, environment, and so on) of initiating transaction of the trace.

Reviewer: Would replace this with a concrete example: If you have a frontend talking to a backend, you can sample the trace by the SDK context available in the frontend, not the backend data.

> When sampling traces the decision on whether to keep or throw away a trace needs to be taken when the first transaction of the trace is seen. This in turn means that for transaction traces only the attributes of the first transaction can participate in the sampling decision.
> 
> In conclusion all trace rules use the attributes (release, environment, and so on) of initiating transaction of the trace.

Author: ```suggestion
When sampling traces, the decision to keep or throw away a trace needs to be taken when the first transaction of the trace is seen. This in turn means that for transaction traces, only the attributes of the first transaction can participate in the sampling decision. For example, if you have a frontend and a backend in the trace, you can only sample the trace by the SDK context available in the frontend, not the backend data.
```

## Changing trace attributes in secondary transactions

Related to the previously discussed issue about trace rules, setting the transaction, the release or the environment in a transaction will have no influence on a trace sampling decision if the transaction is not the first transaction in the trace.

## Limited number of rule conditions

> We're actively working on adding more conditions, from which you can sample by, for example; transaction name. However, feel free to provide suggestions by joining our Discord Server.

Reviewer: link to discord server?

Author: > link to discord server?

@jan-auer - I took this content from an older PR. Is there an actual Discord Server where this discussion is happening? I didn't find anything.

> We're actively working on adding more conditions, from which you can sample by, for example; transaction name. However, feel free to provide suggestions by joining our Discord Server.

Author: ```suggestion
We're actively working on adding more conditions that you can sample by (for example, transaction name). However, feel free to provide suggestions by joining our Discord Server.
```

## Arbitrary logical composition for rule conditions

> At the moment rules containing more than one condition use a logical `and` for each condition. Multiple values for a condition use a logical or condition for matching.
> 
> The decision to implement it like that was taken from the desire to keep the UI simple while still being able to cover all common scenarios.
> 
> Having multiple arguments within a single condition use the or operator is what we commonly want since an attribute generally has one value and having an `and` condition would not make a lot of sense (for example, the environment attribute could be `prod` or `dev` but it could not be prod and dev at the same time).

Author: ```suggestion
Currently, rules containing more than one condition use a logical `and` for each condition. Multiple values for a condition use a logical `or` condition for matching.

The decision to implement it this way was based on the desire to keep the UI simple while still being able to cover all common scenarios.

Having multiple arguments within a single condition use the `or` operator is what we typically want since an attribute generally has one value and having an `and` condition wouldn't make sense. For example, the environment attribute could be `prod` or `dev`, but it couldn't be `prod` and `dev` at the same time.
```

---
title: Current Limitations
sidebar_order: 20
description: "Learn about the current limitions of server-side sampling."
---

<Note>

This feature is in beta and is only available if your organization is participating in its limited release. Features in beta are still in-progress and may have bugs. We recognize the irony. If you’re interested in participating, [join the waitlist](https://sentry.io/for/performance/#updates-signup).

</Note>

Following are the current known limitations and functionality specifics that can create challenges when you're writing server-side sampling rules.

## Supported SDKs & Versions

Server-side sampling is currently only available for the following SDK's:

- Python: 1.8.0 or later
- JavaScript: 7.5.1 or later
- Apple: 7.23.0 or later

> - Android: 6.2.1 or later

Reviewer 2: These versions are inconsistent with the ones on the getting started page.

Reviewer 2: ```suggestion
- Python: 1.7.2 or later
- JavaScript: 7.6.0 or later
- Apple: 7.23.0 or later
- Android: 6.2.1 or later
```

If your application relies on any Sentry SDK that isn't specified above, then you won’t be able to use server-side sampling.

## Trace rules can only select based on the attributes of the initial transaction

When sampling traces, the decision to keep or throw away a trace needs to be taken when the first transaction of the trace is seen. This in turn means that for transaction traces, only the attributes of the first transaction can participate in the sampling decision. For example, if you have a frontend and a backend in the trace, you can only sample the trace by the SDK context available in the frontend, not the backend data.

## Changing trace attributes in secondary transactions

Related to the previously discussed issue about trace rules, setting the transaction, the release, or the environment in a transaction will have no influence on a trace sampling decision if the transaction is not the first transaction in the trace.

## Limited number of rule conditions

> We're actively working on adding more conditions that you can sample by (for example, transaction name). However, feel free to provide suggestions by joining our Discord Server.

Reviewer 3: ```suggestion
We're actively working on adding more conditions that you can sample by (for example, transaction name). However, feel free to provide suggestions by joining the channel [#server-side-sampling](https://discord.com/channels/621778831602221064/1004667579035570176) on our Discord Server.
```

---
title: Current Limitations
sidebar_order: 20
description: "Learn about the current limitions of server-side sampling."
---

<Note>

> This feature is in beta and is only available if your organization is participating in its limited release. Features in beta are still in-progress and may have bugs. We recognize the irony. If you’re interested in joining this limited release, please reach out at [abc@sentry.io](mailto:abc@sentry.io).

Author: ```suggestion
This feature is in beta and is only available if your organization is participating in its limited release. Features in beta are still in-progress and may have bugs. We recognize the irony. If you’re interested in participating, [join the waitlist](https://sentry.io/for/performance/#updates-signup).
```

</Note>

Following are the current known limitations and functionality specifics that can create challenges when you're writing server-side sampling rules.

## Supported SDKs & Versions

Server-side sampling is currently only available for the following SDK's:

- Python: 1.8.0 or later
- JavaScript: 7.5.1 or later
- Apple Cocoa: 7.3.0 or later

> - Java and Android: 5.2.0 or later

Author: ```suggestion
- Java and Android: 6.2.1 or later
```

> <!-- TO DO version numbers need to be confirmed/updated -->

Author: ```suggestion
<!-- TO DO Apple version number needs to be confirmed/updated -->
```

If your application relies on any Sentry SDK that isn't specified above, then you won’t be able to use server-side sampling.

## Trace rules can only select based on the attributes of the initial transaction

> When sampling traces, the decision to keep or throw away a trace needs to be taken when the first transaction of the trace is seen. This in turn means that for transaction traces, only the attributes of the first transaction can participate in the sampling decision. For example, if you have a frontend and a backend in the trace, you can only sample the trace by the SDK context available in the frontend, not the backend data.

Author: @ale-cota - The change I made per our conversation.
